suffuse/root.go

```diff
@@ -0,0 +1,86 @@
+package suffuse
+
+import (
+  "os"
+  "bazil.org/fuse"
+)
+
+/** An interface encapsulating the essential fuse operations.
+ *  Each method returns nil if inapplicable. The fuse operations
+ *  use the first non-nil return value found for a given Path.
+ */
+type Rule interface {
+  MetaData(Path)*fuse.Attr
+  FileData(Path)[]byte
+  DirData(Path)[]fuse.Dirent
+  LinkData(Path)*Path
+}
+
+/** Eventually we'll read this kind of information out of a config
+ *  file, but for now it's hardcoded here.
+ */
+var Rules = []Rule {
+  IdRule{},  // The identity fs - passes underlying paths through unchanged
```

Comment:
Not the way it's implemented now, no - it returns the first match, the other rules get no chance at it. This problem is far-reaching and cannot be solved here: filesystem composition is order dependent and there's no a priori correct way. I think we should remain focused on the behavior of individual filesystems and save composition problems for later. They are only bundled right now for convenience.